### PR TITLE
Hard deprecate old download APIs

### PR DESCRIPTION
**Pull request by ibrahimrabab**
Addresses the following issue: #28627 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java [hunk 1]
```diff
@@ -1040,29 +1040,10 @@ Mono<Response<String>> copyFromUrlWithResponse(BlobCopyFromUrlOptions options, C
...
-     * <p>For more information, see the
-     * <a href="https://docs.microsoft.com/rest/api/storageservices/get-blob">Azure Docs</a></p>
-     *
-     * <p>This method will be deprecated in the future. Use {@link #downloadStream()} instead.
-     *
      * @return A reactive response containing the blob data.
+     * @deprecated use {@link #downloadStream()} instead.
      */
-    @ServiceMethod(returns = ReturnType.COLLECTION)
+    @Deprecated
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java:1046]: I'd continue to have the `@ServiceMethod` annotation even though this has been deprecated. Same comment applies to the other changes like this

**Review by alzimmermsft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java [hunk 2]
```diff
@@ -1040,29 +1040,10 @@ Mono<Response<String>> copyFromUrlWithResponse(BlobCopyFromUrlOptions options, C
...
-     *         downloadData.write&#40;piece.array&#40;&#41;&#41;;
-     *     &#125; catch &#40;IOException ex&#41; &#123;
-     *         throw new UncheckedIOException&#40;ex&#41;;
-     *     &#125;
-     * &#125;&#41;;
-     * </pre>
-     * <!-- end com.azure.storage.blob.specialized.BlobAsyncClientBase.download -->
-     *
-     * <p>For more information, see the
-     * <a href="https://docs.microsoft.com/rest/api/storageservices/get-blob">Azure Docs</a></p>
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java:1059]: I think we should keep this. Leaving samples that are embedded in javadocs for those APIs is OK.
We should check if there are other samples that might be using deprecated apis and update them.

File: sdk/storage/azure-storage-blob/CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 ### Bugs Fixed
 
 ### Other Changes
+- Deprecated BlobClientBase.download and BlobClientBase.downloadWithResponse that downloads entire blob into an output stream.
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/CHANGELOG.md:14]: might be good to mention what's replacement.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java [hunk 4]
```diff
@@ -1040,29 +1040,10 @@ Mono<Response<String>> copyFromUrlWithResponse(BlobCopyFromUrlOptions options, C
...
-     *
-     * <p>This method will be deprecated in the future. Use {@link #downloadStream()} instead.
-     *
      * @return A reactive response containing the blob data.
+     * @deprecated use {@link #downloadStream()} instead.
      */
-    @ServiceMethod(returns = ReturnType.COLLECTION)
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java:1065]: we should retain `ServiceMethod` annotations.

**Pull request closed without merging**